pkg/services/ngalert/models/testing.go

```diff
@@ -601,3 +612,104 @@ func AlertInstanceGen(mutators ...AlertInstanceMutator) *AlertInstance {
 	}
 	return instance
 }
+
+type Mutator[T any] func(*T)
+
+// CopyNotificationSettings creates a deep copy of NotificationSettings.
+func CopyNotificationSettings(ns NotificationSettings, mutators ...Mutator[NotificationSettings]) NotificationSettings {
+	c := NotificationSettings{
+		Receiver: ns.Receiver,
+	}
+	if ns.GroupWait != nil {
+		c.GroupWait = util.Pointer(*ns.GroupWait)
+	}
+	if ns.GroupInterval != nil {
+		c.GroupInterval = util.Pointer(*ns.GroupInterval)
+	}
+	if ns.RepeatInterval != nil {
+		c.RepeatInterval = util.Pointer(*ns.RepeatInterval)
+	}
+	if ns.GroupBy != nil {
+		c.GroupBy = make([]string, len(ns.GroupBy))
+		copy(c.GroupBy, ns.GroupBy)
+	}
+	if ns.MuteTimeIntervals != nil {
+		c.MuteTimeIntervals = make([]string, len(ns.MuteTimeIntervals))
+		copy(c.MuteTimeIntervals, ns.MuteTimeIntervals)
+	}
+	for _, mutator := range mutators {
+		mutator(&c)
+	}
+	return c
+}
+
+// NotificationSettingsGen generates NotificationSettings using a base and mutators.
+func NotificationSettingsGen(mutators ...Mutator[NotificationSettings]) func() NotificationSettings {
+	return func() NotificationSettings {
+		c := NotificationSettings{
+			Receiver:          util.GenerateShortUID(),
+			GroupBy:           []string{model.AlertNameLabel, FolderTitleLabel, util.GenerateShortUID()},
+			GroupWait:         util.Pointer(model.Duration(time.Duration(rand.Intn(100) + 1))),
+			GroupInterval:     util.Pointer(model.Duration(time.Duration(rand.Intn(100) + 1))),
+			RepeatInterval:    util.Pointer(model.Duration(time.Duration(rand.Intn(100) + 1))),
+			MuteTimeIntervals: []string{util.GenerateShortUID(), util.GenerateShortUID()},
+		}
+		for _, mutator := range mutators {
+			mutator(&c)
+		}
+		return c
+	}
+}
+
+type NSMuts struct{}
```

Comment:
I am a little uncomfortable with such names. Exporting abbreviations, even in tests is strange. Let's rename it to NotificationSettingsMutators and then use short variables in tests.